packages/flutter/lib/src/material/material_app.dart

```diff
@@ -73,6 +73,8 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
   GlobalObjectKey _navigator;
 
   Size _size;
+  double _devicePixelRatio;
```

Comment:
I would say we should probably handle size and the other metrics consistently in the didChangeMetrics callback, so if we pull out the other arguments I'd say let's pull out size as well.  (MateralApp's state can still store _size if it wants.)